scipy/optimize/_differentialevolution.py

```diff
@@ -233,6 +244,19 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
     >>> result.x, result.fun
     (array([1., 1., 1., 1., 1.]), 1.9216496320061384e-19)
 
+    Let's try and do a constrained minimization
+    >>> from scipy.optimize import NonlinearConstraint
+    >>> def constr_f(x):
+    ...     return np.array(x[0] + x[1])
+    >>>
+    >>> # the sum of x[0] and x[1] must be less than 1.9
+    >>> nlc = NonlinearConstraint(constr_f, -np.inf, 1.9)
+    >>> bounds = [(0,2), (0, 2)]
```

Comment:
Could you use ``Bounds`` here?
https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.Bounds.html#scipy.optimize.Bounds

I think we should try to use the new classes in the documentation

Reply:
`differential_evolution` has a `bounds` positional argument which can either take the traditional bounds tuple list, or the new style `Bounds` (only added in #9945). The `bounds` positional argument always applies, any `constraints` are applied on top of that. However, to answer your point, yes `Bounds` can be used here.

Reply:
Modified that example to use a `Bounds` object.